packages/ui-app/src/InputNumber.tsx

```diff
@@ -0,0 +1,58 @@
+// Copyright 2017-2018 @polkadot/ui-app authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { BareProps } from './types';
+
+import React from 'react';
+
+import classes from './util/classes';
+import Input from './Input';
+
+type Props = BareProps & {
+  defaultValue?: string,
+  error?: React.ReactNode,
+  info?: React.ReactNode,
+  isError?: boolean,
+  label?: any,
+  maxLength?: number,
+  onChange: (value: string) => void,
+  onKeyDown?: (event: React.KeyboardEvent<Element>) => void,
+  onKeyUp?: (key: string) => void,
+  placeholder?: string,
+  warn?: React.ReactNode,
+  withLabel?: boolean
+};
+
+class InputNumber extends React.PureComponent<Props> {
```

Comment:
Ok, looking this as an "InputNumber". It is just a text input. As a number input, I would expect it to only take numbers. So the logic we have in Balance that does actual numbers should be in here. The only addition Balance should do is to convert to a Param.

1. This component should do number input checks
2. This component should convert to a number (BN in our case)
3. Users of this can further enhance the values

So it is just a move of logic to the base component, i.e. this one.